source/Account.cpp

```diff
@@ -369,12 +369,18 @@ const vector<Mortgage> &Account::Mortgages() const
 
 
 
-// Add a new mortgage for the given amount, with an interest rate determined by
-// your credit score.
-void Account::AddMortgage(int64_t principal)
+// Add a new mortgage for the given amount with the given details. If no interest rate
+// is given then the player's credit score is used to determine the interest rate.
+// If addCredits is true then the principal is also added to the player's credits;
+// otherwise, this is just debt.
+void Account::AddMortgage(int64_t principal, optional<double> interest, int term, bool addCredits)
 {
-	mortgages.emplace_back(principal, creditScore);
-	credits += principal;
+	if(interest)
+		mortgages.emplace_back(principal, *interest, term);
+	else
+		mortgages.emplace_back(principal, creditScore, term);
```

Comment:
I don't think that'll work. Interest and credit score aren't interchangeable. Won't optional<double>.value_or(val) always return a double, therefore calling the wrong constructor when we want the credit score constructor instead of the interest rate constructor?

Reply:
F you're absolutely correct, my bad